tests/multi_agents/test_supervisor_multi_agent.py

```diff
@@ -0,0 +1,97 @@
+from unittest.mock import AsyncMock, MagicMock
+
+import pytest
+from langchain_core.language_models import GenericFakeChatModel
+from langchain_core.messages import HumanMessage, SystemMessage
+
+from neuroagent.multi_agents.supervisor_multi_agent import AgentState
+from src.neuroagent.multi_agents import SupervisorMultiAgent
+
+
+def test_create_main_agent_initialization():
+    mock_llm = MagicMock()
+    bind_function_result = MagicMock()
+    bind_function_result.__ror__.return_value = {}
+    mock_llm.bind_functions.return_value = bind_function_result
+    data = {"llm": mock_llm, "agents": [("agent1",)]}
+
+    result = SupervisorMultiAgent.create_main_agent(data)
+    assert "main_agent" in result
+    assert "summarizer" in result
+
+
+@pytest.mark.asyncio
+async def test_agent_node():
+    mock_message = HumanMessage(
+        content="hello",
+        name="test_agent",
+    )
+
+    async def mock_ainvoke(_):
+        return {"messages": [mock_message]}
+
+    agent_state = MagicMock()
+    agent = MagicMock()
```

Comment:
If a dunder method is not mocked, magic mock can be replaced with a regular mock I believe